src-java/kilda-pce/src/main/java/org/openkilda/pce/impl/InMemoryPathComputer.java

```diff
@@ -266,6 +281,41 @@ private PathWeight weightByCost(Edge edge) {
         return new PathWeight(cost, penalties);
     }
 
+    private PathWeight weightByLatencyForProtectedPath(Edge edge) {
+        Map<Penalty, Long> penalties = new EnumMap<>(Penalty.class);
+
+        if (edge.isUnderMaintenance()) {
+            penalties.put(UNDER_MAINTENANCE, config.getUnderMaintenanceLatencyRaise());
+        }
+
+        if (edge.isUnstable()) {
+            penalties.put(UNSTABLE, config.getUnstableLatencyRaise());
+        }
+
+        if (edge.getDiversityGroupUseCounter() > 0) {
+            long value = edge.getDiversityGroupUseCounter() * config.getDiversityIslLatency();
+            penalties.put(HARD_DIVERSITY_ISL_LATENCY, value);
+        }
+
+        if (edge.getDiversityGroupPerPopUseCounter() > 0) {
+            int value = edge.getDiversityGroupPerPopUseCounter() * config.getDiversityPopIslCost();
+            penalties.put(DIVERSITY_POP_ISL_COST, (long) value);
+        }
+
+        if (edge.getDestSwitch().getDiversityGroupUseCounter() > 0) {
+            long value = edge.getDestSwitch().getDiversityGroupUseCounter() * config.getDiversitySwitchLatency();
+            penalties.put(HARD_DIVERSITY_SWITCH_LATENCY, value);
+        }
+
+        if (edge.getAffinityGroupUseCounter() > 0) {
+            long value = edge.getAffinityGroupUseCounter() * config.getAffinityIslLatency();
+            penalties.put(AFFINITY_ISL_LATENCY, value);
+        }
+
+        long edgeLatency = edge.getLatency() <= 0 ? config.getDefaultIslLatency() : edge.getLatency();
+        return new PathWeight(edgeLatency, penalties);
+    }
```

Comment:
Most of these fields is a copy of `weightByLatency`. Maybe we should add some third method with base penalties:

```
getBasePenalcies(***) {
****
}

private PathWeight weightByLatency(Edge edge) {
penalties = getBasePenalties()
 if (edge.getDiversityGroupUseCounter() > 0) {
            long value = edge.getDiversityGroupUseCounter() * config.getDiversityIslLatency();
            penalties.put(HARD_DIVERSITY_ISL_LATENCY, value);
        }
        ****
}

private PathWeight weightByLatencyForProtectedPath(Edge edge) {
penalties = getBasePenalties()
        if (edge.getDiversityGroupUseCounter() > 0) {
            long value = edge.getDiversityGroupUseCounter() * config.getDiversityIslLatency();
            penalties.put(DIVERSITY_ISL_LATENCY, value);
        }
        ***
}
```